### DISCUSSION TITLE
How to convert to CDXML?

### DISCUSSION THREAD

**Discussion by pa-nana** [rdkit/rdkit#4762] [Q&A] [answered]
Hi,

Our project uses OpenBabel now and we are planning to use RDKit instead. I would like to know if RDKit can convert to cdxml format?

I am using
* python 3.8.12
* rdkit 2021.09.2 (conda-forge)

I found the code https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/FileParsers/mol.py
```
from Chem import rdmol
from Chem.rdmol import Atom, Bond, Mol
...
print(rdmol.MolToCDXML(m))
```

I wondered if I could use `MolToCDXML`, so I tried to do the same thing and I got `ModuleNotFoundError: No module named 'Chem'`. 
I would like to know if I miss any or this is not the correct way to get the cdxml ? Thank you.

**Comment by greglandrum** [comment, marked as answer]
The RDKit does not currently support either reading or writing CDXML.

**Reply by pa-nana** [reply]
Hi @greglandrum , got it, thanks for the quick response!

**Comment by kienerj** [comment]
I probably should write my 2 cents here. I've looked at ChemDraw formats (cdx, cdxml) extensively. They are drawing format with molecules as first-class citizens. Therefore any cdxml -> other format is tricky and will likley loose information. However rdkit to cdxml is entirely doable as I have working python code doing that with a list of caveats obviously. (doable doesn't mean it's easy!)

However said code is for now in a private repository. Ultimately the plan is to make it public but there are unanswered questions which I'm not very keen to invest too much time on.  For example but not limited to the legal side, if "reverse-engineering" the format (the official specification is outdated) is allowed or not and who wants to potentially deal with lawyers from a large org like PerkinElmer?

However if there is certain interest from other parties or the community in this especially also in regards to maintenance once made public (where is the fun once the problem is reasonably solved?...) "joining forces" might be an option. What I advise against trying to do it yourself (at least if you use python) from scratch again.
